tt_eager/tt_lib/csrc/tt_lib_bindings_tensor_pytensor.cpp

```diff
@@ -1313,15 +1313,30 @@ Tensor convert_python_tensors_to_tt_tensors(py::list tensor_shards, std::optiona
                     storage_type = tt_tensor.storage_type()
 
             )doc")
-            .def(
-                "device", [](const Tensor &self) { return self.device(); }, R"doc(
+        .def(
+            "device",
+            [](const Tensor &self) { return self.device(); },
+            R"doc(
                 Get the device of the tensor.
 
                 .. code-block:: python
 
                     device = tt_tensor.device()
 
-            )doc")
+            )doc",
+            py::return_value_policy::reference)
```

Comment:
@aliuTT you might care about this one. Read the original issue on this.

Reply:
were there existing issues that this fixes?

Reply:
yeah, we were observing the device getting destructed